### PR TITLE
Use unified config framework

### PR DESCRIPTION
**Pull request by feihujiang**
Using nconf to adopt a unified configuration system.

### CODE REVIEW

File: src/comm/client/local-client.js [hunk 1]
```diff
@@ -11,14 +11,23 @@
 const bc   = require('../blockchain.js');
 const RateControl = require('../rate-control/rateControl.js');
 const Util = require('../util.js');
+const cfUtil = require('../config-util.js');
+const path = require('path');
+const config = cfUtil.getConfig();
+const default_config = path.resolve(__dirname, '../../../config/default.json');
```
[the remark this answers is not in the document]
Comment (feihujiang) [reply src/comm/client/local-client.js]: Fixed

File: src/comm/client/local-client.js [hunk 2]
```diff
@@ -11,14 +11,23 @@
...
+//make sure this default has precedences
+config.reorderFileStores(default_config);
```
[the remark this answers is not in the document]
Comment (feihujiang) [reply src/comm/client/local-client.js]: Fixed

**Comment by feihujiang** [comment]
@aklenik Changed. Please take a look.

File: src/comm/Config.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+     */
+    constructor() {
+        nconf.use('memory');
+        nconf.use('mapenv', {type:'memory'});
+        this.mapSettings(nconf.stores.mapenv, process.env);
+        this._fileStores = [];
+        // reference to configuration settings
+        this._config = nconf;
+        const defaultConfig = commUtils.resolvePath('config/default.yaml');
+        this.file(defaultConfig);
```
Comment (aklenik) [inline src/comm/Config.js]: This should be moved after the `nconf.env()` line to make it the lowest priority.

Comment (feihujiang) [reply src/comm/Config.js]: Fixed

File: config/default.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
[the remark this answers is not in the document]
Comment (feihujiang) [reply config/default.json]: Removed.

**Comment by feihujiang** [comment]
@aklenik Updated, please take a look.

**Pull request merged**